### DISCUSSION TITLE
Optimized search results

### DISCUSSION THREAD

**Discussion by lharris-te** [GitbookIO/community#54] [Feature Requests]
### What's your feature/idea?

I'd like to be able to "weight" the list of results that GitBook returns when users use Search on our site.

Currently, when users enter a string in Search, GitBook returns _any_ article that uses it. I think GitBook might foreground articles that use the string in an H1, H2, etc. heading.

But we are maintaining a pretty long history of legacy content - for example, release notes from 10 years ago! I'd like to weight search results so that those older relnotes articles appear **last** in the list. 
(Better yet, can I set a date cutoff?)

Another way to look at this is to implement **metadata** for files managed by GitBook: 
Let me set a few keywords for each article and use these keywords in the Search algorithm, so that more important or more recent content shows up first in the list of results.
Some keywords I can imagine using might be
`new_feature`
`upgrade`
`installation_process`
`archived` (I'd use this one to mark articles _not_ to list in search results)

### What's your desired outcome?

Users would see search results that only include relevant content.

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

### Anything else?

Thank you!

**Comment by RensTillmann** [comment]
I need this too. Including an option to exclude a specific page or pages from the results.
